module/mempool/herocache/backdata/heropool/pool_test.go

```diff
@@ -215,6 +217,111 @@ func TestInvalidateEntity(t *testing.T) {
 	}
 }
 
+// TestAddAndRemoveEntities checks health of heroPool for scenario where entitites are stored and removed in a predetermined order.
+// LRUEjection, NoEjection and RandomEjection are tested. RandomEjection doesn't allow to provide a final state of the pool to check.
+func TestAddAndRemoveEntities(t *testing.T) {
+	for _, tc := range []struct {
+		limit               uint32       // capacity of the pool
+		entityCount         uint32       // total entities to be stored
+		ejectionMode        EjectionMode // ejection mode
+		numberOfOperations  int
+		probabilityOfAdding float32
+	}{
+		{
+			limit:               500,
+			entityCount:         1000,
+			ejectionMode:        LRUEjection,
+			numberOfOperations:  1000,
+			probabilityOfAdding: 0.8,
+		},
+		{
+			limit:               500,
+			entityCount:         1000,
+			ejectionMode:        NoEjection,
+			numberOfOperations:  1000,
+			probabilityOfAdding: 0.8,
+		},
+		{
+			limit:               500,
+			entityCount:         1000,
+			ejectionMode:        RandomEjection,
+			numberOfOperations:  1000,
+			probabilityOfAdding: 0.8,
+		},
+	} {
+		t.Run(fmt.Sprintf("%d-limit-%d-entities", tc.limit, tc.entityCount), func(t *testing.T) {
+			testAddRemoveEntities(t, tc.limit, tc.entityCount, tc.ejectionMode, tc.numberOfOperations, tc.probabilityOfAdding)
+		})
+	}
+}
+
+// Indexes contains an index in the pool of an inserted entity and an index of this entity
+// in the array of entities created by the EntityListFixture.
+type Indexes struct {
+	indexInPool        EIndex
+	indexInEntitiesArr int
+}
+
+func testAddRemoveEntities(t *testing.T, limit uint32, entityCount uint32, ejectionMode EjectionMode, numberOfOperations int, probabilityOfAdding float32) {
```

Comment:
Please add a `godoc` explaining what does this test do and what the parameters are.

Reply:
sure, good that you noticed ! my bad